geomesa-spark/geomesa-spark-core/src/main/scala/org/locationtech/geomesa/spark/GeoMesaSpark.scala

```diff
@@ -8,32 +8,76 @@
 
 package org.locationtech.geomesa.spark
 
+import java.io.{BufferedWriter, StringWriter}
 import java.util.ServiceLoader
 
 import org.apache.hadoop.conf.Configuration
 import org.apache.spark.SparkContext
 import org.apache.spark.rdd.RDD
 import org.geotools.data.Query
-import org.opengis.feature.simple.SimpleFeature
+import org.geotools.geojson.feature.FeatureJSON
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.collection.JavaConversions._
+import scala.reflect._
 
 trait SpatialRDDProvider {
+  import TypeDefault._
+
+  type GeoJSONString = String
+
+  def toScalaMap(in: SpatialRDD): RDD[Map[String, AnyRef]] = {
```

Comment:
we should make this private or protected since looks like they aren't meant to be called publicly